### PR TITLE
application-title-bar: init at 0.6.2

### PR DESCRIPTION
**Pull request by HeitorAugustoLN**
## Description of changes

- Added HeitorAugustoLN (me) to the maintainers list
- Added a new package application-title-bar, a KDE Plasma 6 Widget

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by pbsds** [comment]
please remove the merge commits

**Comment by HeitorAugustoLN** [comment]
I don't know why it happened, but now it is removed

File: maintainers/maintainer-list.nix
Comment (pbsds) [inline maintainers/maintainer-list.nix:1]: please split of this into a separate commit titled `maintainers: add HeitorAugustoLN`

Comment (HeitorAugustoLN) [reply maintainers/maintainer-list.nix:1]: Done.

File: pkgs/by-name/ap/application-title-bar/package.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, pkgs
```
Comment (sikmir) [inline pkgs/by-name/ap/application-title-bar/package.nix:4]:
```suggestion
, kdePackages
```

File: pkgs/by-name/ap/application-title-bar/package.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  version = "0.6.2";
+
+  src = fetchFromGitHub {
+    owner = "antroids";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-KKODCfLL+T4NdflxUFP++TO3OtQWJpLHAnUVqmvIzrg=";
+  };
+
+  buildInputs = with pkgs.kdePackages; [qttools kconfig];
```
Comment (sikmir) [inline pkgs/by-name/ap/application-title-bar/package.nix:18]:
```suggestion
  buildInputs = with kdePackages; [ qttools kconfig ];
```

File: pkgs/by-name/ap/application-title-bar/package.nix (hunk 1 above)
Comment (HeitorAugustoLN) [reply pkgs/by-name/ap/application-title-bar/package.nix]: Done

File: pkgs/by-name/ap/application-title-bar/package.nix (hunk 2 above)
Comment (HeitorAugustoLN) [reply pkgs/by-name/ap/application-title-bar/package.nix]: Done

File: pkgs/by-name/ap/application-title-bar/package.nix [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  buildInputs = with kdePackages; [ qttools kconfig ];
+
+  dontWrapQtApps = true;
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/{bin,share/plasma/plasmoids/com.github.antroids.application-title-bar}
+    cp -r $src/package/* $out/share/plasma/plasmoids/com.github.antroids.application-title-bar
+    ln -s ${kdePackages.kconfig}/bin/{kwriteconfig6,kreadconfig6} $out/bin
+    ln -s ${kdePackages.qttools}/bin/qdbus $out/bin
```
Comment (pbsds) [inline pkgs/by-name/ap/application-title-bar/package.nix:27]: instead of symlinking these to /bin/, which might interfere with a FHS user env, have you tried adding these packages to `propagatedUserEnvPkgs` or `propagatedBuildInputs`?

File: pkgs/by-name/ap/application-title-bar/package.nix [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+, kdePackages
+}:
+
+stdenv.mkDerivation rec {
+  pname = "application-title-bar";
+  version = "0.6.2";
+
+  src = fetchFromGitHub {
+    owner = "antroids";
+    repo = pname;
```
Comment (SuperSandro2000) [inline pkgs/by-name/ap/application-title-bar/package.nix:13]:
```suggestion
    repo = "application-title-bar";
```

File: pkgs/by-name/ap/application-title-bar/package.nix (hunk 3 above)
Comment (SuperSandro2000) [reply pkgs/by-name/ap/application-title-bar/package.nix:27]: qttools should also be part of plasma already

Comment (HeitorAugustoLN) [reply pkgs/by-name/ap/application-title-bar/package.nix:27]: > instead of symlinking these to /bin/, which might interfere with a FHS user env, have you tried adding these packages to `propagatedUserEnvPkgs` or `propagatedBuildInputs`?

I've tried propagatedBuildInputs, but it did not work back when I tried, but I tried propagatedUserEnvPkgs now, and it works.

**Comment by pbsds** [comment]
please squash down your commits to the following two in this order:

* `maintainers: add HeitorAugustoLN`
* `application-title-bar: init at 0.6.2`

**Comment by HeitorAugustoLN** [comment]
> please squash down your commits to the following two in this order:
> 
> * `maintainers: add HeitorAugustoLN`
> * `application-title-bar: init at 0.6.2`

Done.
